mobile/examples/phi-3/android/app/src/main/java/ai/onnxruntime/genai/demo/MainActivity.java

```diff
@@ -84,9 +108,44 @@ public void onClick(View v) {
                 new Thread(new Runnable() {
                     @Override
                     public void run() {
+                        TokenizerStream stream = null;
+                        GeneratorParams generatorParams = null;
+                        Generator generator = null;
+                        Sequences encodedPrompt = null;
                         try {
-                            genAIWrapper.run(promptQuestion_formatted);
-                        } catch (GenAIException e) {
+                            stream = tokenizer.createStream();
+
+                            generatorParams = model.createGeneratorParams();
+                            //examples for optional parameters to format AI response
+                            //generatorParams.setSearchOption("length_penalty", 1000);
+                            //generatorParams.setSearchOption("max_length", 500);
+
+                            encodedPrompt = tokenizer.encode(promptQuestion_formatted);
+                            generatorParams.setInput(encodedPrompt);
+
+                            generator = new Generator(model, generatorParams);
+
+                            while (!generator.isDone()) {
+                                generator.computeLogits();
+                                generator.generateNextToken();
+                 
+                                int token = generator.getLastTokenInSequence(0);
+                 
+                                tokenListener.accept(stream.decode(token));
+                            }
+
+                            generator.close();
+                            encodedPrompt.close();
+                            stream.close();
+                            generatorParams.close();
+
+                        }
+                        catch (GenAIException e) {
+                            Log.e(TAG, "Exception occurred during model query: " + e.getMessage());
+                            if (generator != null) generator.close();
+                            if (encodedPrompt != null) encodedPrompt.close();
+                            if (stream != null) stream.close();
+                            if (generatorParams != null) generatorParams.close();
                             throw new RuntimeException(e);
```

Comment:
Do we need to convert this to RuntimeException or could we just rethrow?

Reply:
just followed existing syntax on previous example.